tests/_temporary_private_container_api_test.py

```diff
@@ -87,3 +87,58 @@ def test_paths_outside_private_path_conversion(
         get_incontainer_path(typed_host_path, typed_host_runner_path)
 
     assert isinstance(raised_exc_info.value.__cause__, ValueError)
+
+
+@pytest.mark.parametrize(
+    (
+        'host_path',
+        'host_runner_path',
+        'container_root',
+        'expected_incontainer_path',
+    ),
+    (
+        ('/root', '/', '/tmp', '/tmp/root'),
+        (
+            '/tmp/private/subdir',
+            '/tmp/private',
+            '/tmp/private',
+            '/tmp/private/subdir',
+        ),
+        (
+            '/tmp/private/subdir',
+            '/tmp/private',
+            None,
+            f'{CONTAINER_ROOT}/subdir',
+        ),
+    ),
+    ids=('differing-paths', 'same-paths', 'default-container-root'),
+)
+@pytest.mark.parametrize(
+    'convert_path_to_type',
+    (pathlib.Path, str),
+    ids=('host-path-pathlib', 'host-path-str'),
+)
+@pytest.mark.parametrize(
+    'convert_runner_path_to_type',
+    (pathlib.Path, str),
+    ids=('host-path-pathlib', 'host-path-str'),
+)
+# pylint: disable-next=too-many-arguments,too-many-positional-arguments
+def test_provide_custom_container_root(  # noqa: WPS211
```

Comment:
@fosterseth could you look into adding similar coverage for `None` vs. path for `inject_credential`?